node.go

```diff
@@ -69,9 +87,20 @@ func (node *node) getChannel() (*channel, error) {
 		PermitWithoutStream: true,
 	}
 
-	conn, err := grpc.Dial(node.address, grpc.WithInsecure(), grpc.WithKeepaliveParams(kacp), grpc.WithBlock())
+	var conn *grpc.ClientConn
+	var err error
+	parts := strings.SplitN(node.address, ":", 2)
+	if parts[1] == "443" || parts[1] == "50212" {
+		cont, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
```

Comment:
I could be wrong, but I think calling `defer cancel()` will call `cancel` as soon as this variable goes out of scope of the function which would immediately cancel the connection.

Reply:
i am not sure, but if i don't have it, it will not stop running even if timeout is reached for some reason